src/main/java/service/BookServiceImpl.java

```diff
@@ -1,16 +1,35 @@
 package service;
 
+import dto.BookDto;
+import dto.CreateBookRequestDto;
+import exception.EntityNotFoundException;
 import java.util.List;
+import mapper.BookMapper;
 import model.Book;
+import repository.BookRepository;
 
 public class BookServiceImpl implements BookService {
+    private BookRepository bookRepository;
+    private BookMapper bookMapper;
```

Comment:
I do not see where the initialisation of this fields is done. Also, good candidates to be final